pkg/registry/core/service/storage/transaction.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+// transaction represents something that may need to be finalized on success or
+// failure of the larger transaction.
+type transaction interface {
```

Comment:
it is an operation really. not transaction. I fear that people will get the wrong ideas about the idempotency/acid properties of "a transaction". e.g. COMMIT may not work completely.

Reply:
"operation" doesn't capture the semantics I want.  "pending"?  "committable"?  I don't care much, I thought txn was close enough